src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separationString = ',') {
+    let joinedString = '';
+    const separator = String(separationString);
```

Comment:
when you concatenate a separator with a string, it will be converted to a string anyway